actors/rozetka-count/src/main.js

```diff
@@ -0,0 +1,115 @@
+    });
+
+    /**
+     * At start we check whether the key-value store has a value
+     * of the STATS_KEY key (in case previous actor run migrated here)
+     * and save the value as a start value for the current run.
+     */
+    const statsObj = await Apify.getValue(STATS_KEY);
+    if (typeof statsObj === 'object' && statsObj != null) {
+        const stats = statsObj;
+
+        log.info(`Product count start value from the store: ${stats.value}`)
+        await getOrIncStatsValue(+stats.value);
+    }
+
+    // persist value on migration
+    Apify.events.on('migrating', async () => {
+        migrationFlagGetOrSet(true);
+        const statsValue = await getOrIncStatsValue();
+
+        log.info(`Migration: saving the value '${statsValue}' to the key-value store.`);
+        await Apify.setValue(STATS_KEY, { value: statsValue });
+    })
+
+    setInterval(async () => log.info(`Shop products count: ${await getOrIncStatsValue()}`), 60000);
+
+    const crawler = new Apify.CheerioCrawler({
+        requestList,
+        requestQueue,
+        proxyConfiguration,
+        useSessionPool: true,
+        persistCookiesPerSession: true,
+        maxRequestRetries: 5,
+        handlePageTimeoutSecs: 120,
+        requestTimeoutSecs: 120,
+        autoscaledPoolOptions: {
+            maxConcurrency,
+        },
+        maxRequestsPerCrawl,
+        handlePageFunction: async (context) => {
+            const { request } = context;
+            const { url, userData } = request;
+
+            const { label } = userData;
+            log.info('Page opened.', { label, url });
+            switch (label) {
+                case MAIN_PAGE:
+                    await handleMainPage(context);
+                    break;
+                case CATEGORY:
+                    await handleCategory(context);
+                    break;
+                case PRODUCT_LIST:
+                    await handleProductList(context);
+                    break;
+                case RESTAURANT_CATEGORY:
+                    await handleRestaurantCategory(context);
+                    break;
+                case RESTAURANT_SUBCATEGORY:
+                    await handleRestaurantSubcategory(context);
+                    break;
+                case CLOTHES_CATEGORY:
+                    await handleClothesCategory(context);
+                    break;
+
+                default:
+                    break;
+            }
+        },
+        handleFailedRequestFunction: async ({ request }) => {
+            const debugDataSet = await Apify.openDataset('debug-rozetka-count');
+            await debugDataSet.pushData({
+                '#debug': Apify.utils.createRequestDebugInfo(request),
+            });
+        },
+    });
+
+    log.info('Starting the crawl.');
+    await crawler.run();
+    log.info('Crawl finished.');
+
+    await Apify.pushData({ OUTPUT: await getOrIncStatsValue() })
+    await Apify.setValue(STATS_KEY, null);
```

Comment:
Why you save a null here?

Reply:
I use a named store to preserve the results of the counting in case of migration of the docker container. So that, when the migrated container restarts, it takes the initial count value from the store. That's why I clear this store when the crawling is completed, so that another separate crawling cycle doesn't mix up with the previous actor's run resulting count value. Is there a better solution for this or is there any dangers with this approach?

Reply:
If it is just for migrated containers, then use the default KV store. Then you don't need to clean it up because it is tied to the run. Named KV is only for saving between multiple runs (migration is still the same run).

Reply:
That sounds perfect. Thanks a lot for the advise!